akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEventParser.scala

```diff
@@ -22,6 +22,58 @@ import scala.util.control.NonFatal
 
 private object ServerSentEventParser {
 
+  private class Builder {
+
+    private var data = Vector.empty[String]
+
+    private var eventType = null: String
+
+    private var id = null: String
+
+    private var retry = null: String
+
+    private var _size = 0
+
+    def appendData(value: String): Unit = {
+      data :+= value
+      _size += 5 + value.length
+    }
+
+    def setEventType(value: String): Unit = {
+      eventType = value
+      _size += 6 + value.length
+    }
+
+    def setId(value: String): Unit = {
+      id = value
```

Comment:
For the parser it's probably more important to look at the memory size, not at the wire size. Hence you are right.